frontend/src/components/pages/CreateReview/BookModal.tsx

```diff
@@ -0,0 +1,275 @@
+import { CheckIcon, SmallCloseIcon } from "@chakra-ui/icons";
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Grid,
+  GridItem,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+
+import { BookFormats, Genre } from "../../../constants/Enums";
+import { Author, Book, BookFormat, Publisher } from "../../../types/BookTypes";
+import AddNumberInput from "./AddNumberInput";
+import AddSelect from "./AddSelect";
+import AddStringInput from "./AddStringInput";
+import AddStringInputList from "./AddStringInputList";
+
+interface BookModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  booksAdded: Book[];
+  handleBooksAdded: (b: Book[]) => void;
+}
+
+const BookModal = (props: BookModalProps): React.ReactElement => {
+  const { isOpen, onClose, booksAdded, handleBooksAdded } = props;
+
+  // Book Fields
+  const [title, setTitle] = useState<string>("");
+  const [coverImage, setCoverImage] = useState<string>("");
+  const [titlePrefix, setTitlePrefix] = useState<string>("");
+  const [seriesOrder, setSeriesOrder] = useState<string>("");
+  const [illustrators, setIllustrator] = useState<string[]>([]);
+  const [translators, setTranslator] = useState<string[]>([]);
+  const [format, setFormat] = useState<string>("");
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [genre, setGenre] = useState<string>("");
+  const [minAge, setMinAge] = useState<number>(0);
+  const [maxAge, setMaxAge] = useState<number>(0);
+  const [authors, setAuthors] = useState<string[]>([]);
+  const [publisher, setPublisher] = useState<string>("");
+  const [seriesName, setSeriesName] = useState<string>("");
+  const [isbn, setIsbn] = useState<string>("");
+  const [price, setPrice] = useState<number>(0);
+  const [publishYear, setPublicationYear] = useState<number>(0);
+
+  const updateBookObj = () => {
+    const authorObjs: Author[] = [];
+
+    authors.forEach((author) => {
+      authorObjs.push({
+        fullName: author,
+        displayName: author,
```

Comment:
Why are both fullName and displayName just the entered value of the authors? If there is no difference, why do we have both. If they are different, where can we add different values for fullname and display name? Sorry for so many questions lol. @susheelk maybe you can help with this?

Reply:
LOL no need to apologize, i asked the same questions as you! 

So the reason we have both is b/c some people want to have a diff display name i guess? Like S.E. Hinton vs Susan Eloise Hinton, who wanted to hide her first name. I also wasn't sure where to input different values b/n full-name and display name so it's gonna be something we'll determine down the road whether or not we need both.